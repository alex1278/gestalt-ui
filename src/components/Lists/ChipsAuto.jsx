import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Autocomplete } from 'react-md';
import { Error } from 'components/Typography';
import { insertItem, removeItem } from 'util/helpers/lists';
import Chip from './Chip';

const List = styled.div`
  display: flex;
  flex-wrap: wrap;
`;

class Chips extends Component {
  static propTypes = {
    meta: PropTypes.object.isRequired,
    input: PropTypes.object,
    label: PropTypes.string,
    data: PropTypes.array,
    helpText: PropTypes.string,
  };

  static defaultProps = {
    input: {},
    label: 'Entry',
    data: [],
    helpText: null,
  };

  state = { touched: false };

  addItem = (value) => {
    const isDupe = this.props.input.value && this.props.input.value.find(item => item === value);

    if (value && !isDupe) {
      this.props.input.onChange(insertItem(this.props.input.value, value));
    }
  }

  removeItem = (value) => {
    this.props.input.onChange(removeItem(this.props.input.value, value));
  }

  handleTouched = () => {
    this.setState({ touched: true });
  }

  render() {
    const { label, data, helpText, input, meta: { error }, ...rest } = this.props;

    return (
      <React.Fragment>
        <Autocomplete
          id={`${label}-chips`}
          placeholder={this.props.label}
          fullWidth
          onAutocomplete={this.addItem}
          lineDirection="center"
          onFocus={this.handleTouched}
          filter={Autocomplete.caseInsensitiveFilter}
          data={data}
          clearOnAutocomplete
          focusInputOnAutocomplete
          listInline
          helpText={helpText}
          {...rest}
        />
        <List maxHeight="184px">
          {Array.isArray(input.value) && input.value.map((item, i) => <Chip key={`${item}--${i}`} item={item} onRemove={this.removeItem} />)}
        </List>
        {(this.state.touched && !!error) && <Error>{error}</Error>}
      </React.Fragment>
    );
  }
}

export default Chips;
