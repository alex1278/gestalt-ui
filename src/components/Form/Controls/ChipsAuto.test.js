import React from 'react';
import { render } from 'react-testing-library';
import ChipsAuto from './MultiSelect';

describe('ChipsAuto', () => {
  it('mounts with basic props', () => {
    const { container } = render(<ChipsAuto meta={{ error: null, touched: false }} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
