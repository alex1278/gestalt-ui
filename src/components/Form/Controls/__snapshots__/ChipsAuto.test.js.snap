// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChipsAuto mounts with basic props 1`] = `
<div
  class="css-a0jhvo"
  id="Entry-chips"
>
  <div
    class="MuiFormControl-root-4 MuiFormControl-fullWidth-7"
  >
    <div
      class="MuiInputBase-root-21 MuiOutlinedInput-root-8 MuiInputBase-fullWidth-30 MuiInputBase-formControl-22"
    >
      <fieldset
        aria-hidden="true"
        class="MuiPrivateNotchedOutline-root-38 MuiOutlinedInput-notchedOutline-15"
        style="padding-left: 8px;"
      >
        <legend
          class="MuiPrivateNotchedOutline-legend-39"
          style="width: 0.01px;"
        >
          <span>
            â€‹
          </span>
        </legend>
      </fieldset>
      <div
        aria-invalid="false"
        class="MuiInputBase-input-31 MuiOutlinedInput-input-16 IntegrationReactSelect-input-1"
        type="text"
      >
        <div
          class="css-1hwfws3"
        >
          <div
            class="css-1492t68"
          >
            Entry
          </div>
          <div
            class="css-1tna5xz"
          >
            <div
              class=""
              style="display: inline-block;"
            >
              <input
                aria-autocomplete="list"
                autocapitalize="none"
                autocomplete="off"
                autocorrect="off"
                id="react-select-2-input"
                spellcheck="false"
                style="box-sizing: content-box; width: 2px; border: 0px; font-size: inherit; opacity: 1; outline: 0; padding: 0px;"
                tabindex="0"
                type="text"
                value=""
              />
              <div
                style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-size: inherit; letter-spacing: normal; text-transform: none;"
              />
            </div>
          </div>
        </div>
        <div
          class="css-1wy0on6"
        >
          <span
            class="css-d8oujb"
          />
          <div
            aria-hidden="true"
            class="css-1ep9fjw"
          >
            <svg
              aria-hidden="true"
              class="css-19bqh2r"
              focusable="false"
              height="20"
              viewBox="0 0 20 20"
              width="20"
            >
              <path
                d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
